src/main/java/org/folio/consortia/service/impl/UserTenantServiceImpl.java

```diff
@@ -79,4 +124,98 @@ public UserTenantCollection getByUsernameAndTenantId(UUID consortiumId, String u
     result.setTotalRecords(1);
     return result;
   }
+
+  private User prepareShadowUser(UUID userId, UserTenantEntity userTenantEntity, FolioExecutionContext folioExecutionContext) {
+    try (var context = new FolioExecutionContextSetter(prepareContextForTenant(userTenantEntity.getTenant().getId(), folioExecutionContext))) {
+      User user = new User();
+      User userOptional = getUser(userId);
+
+      if (Objects.nonNull(userOptional.getId())) {
+        user.setId(UUID.randomUUID().toString());
+        user.setPatronGroup(PATRON_GROUP);
+        user.setUsername(userOptional.getUsername() + HelperUtils.randomString(RANDOM_STRING_COUNT));
+        if (Objects.nonNull(userOptional.getPersonal())) {
+          Personal personal = new Personal();
+          personal.setLastName(userOptional.getPersonal().getLastName());
```

Comment:
userOptional.getPersonal() can be extracted and reused

Reply:
we need only some specific fields , as user may contain sensitive data.

Reply:
I mean extracting local variable, like this: userPersonal=userOptional.getPersonal() and than use it: personal.setLastName(userPersonal.getLastName()), ... so that you do not have to call this in each line